content/js/record.md

```diff
@@ -1,335 +1,189 @@
+```js
+// define record type "person"
+const PersonRecord = skygear.Record.extend("person");
+// create "person" record: bob
+const bob = new PersonRecord({ name: "bob", age: 20 });
+
+{ // bob object:
+  name: "bob"
+  age: 20
+  // read-only metadata provided by skygear:
+  recordType: "person"
+  id: "person/45b0272b-41dc-483c-a07d-798a95cabbbd"
+  ownerID: "7888d764-a20a-4c2c-89c2-8eb02de0a079"
+  createdAt: Mon Oct 17 2016 16:49:14 GMT+0800 (HKT)
+  createdBy: "7888d764-a20a-4c2c-89c2-8eb02de0a079"
+  updatedAt: Mon Oct 17 2016 16:49:14 GMT+0800 (HKT)
+  updatedBy: "7888d764-a20a-4c2c-89c2-8eb02de0a079"
+}
 ```
+When creating Skygear records, you should avoid starting your type name with `_` and the fields names with either `_` or `$` as they are reserved by Skygear for internal usage. You should also avoid naming your fields the same as the metadata shown above.
 
-### Record Database
-
-You will be provided with a private and a public database.
+<a name="public-private-database"></a>
+## Public/Private Database
+Skygear provides 2 places where you could save records: a `privateDB` that can only be accessed by the user and a `publicDB` that's controlled by access rules attached to each record. By default, every record in the `publicDB` is readable by everyone and only writable by its creator.
 
-- Everything in the private database is truly private, regardless of what access
-control entity you set to the record. In other words, each user has his own
-private database, and only himself has access to it.
-- Record saved at public database is by default public. Even without
-logging in, records in the public database can be queried (but not updated).
-To control the access, you may set different access control to the record.
-- The database objects can be accessed with `skygear.publicDB` and
-`skygear.privateDB`.
+Refer to the [Access Control][] section for more detail.
 
-<a name="basic-crud"></a>
-## Basic CRUD
+<a name="sql-database-backend"></a>
+### SQL Database Backend
+On the Skygear server backend, a relational SQL database is used to store your data. The record's type name is mapped to a table name while field names are mapped to table columns. You can view and modify skygear's data by connecting directly to the database.
 
-### Creating a record
-
-You can save a public record to server as the following.
-
-``` javascript
-skygear.publicDB.save(new Note({
-  'content': 'Hello World!'
-})).then((record) => {
-  console.log(record);
-}, (error) => {
-  console.error(error);
-});
+<a name="save"></a>
+## Save
+You can save skygear records using a single API call, here's how you could save the `bob` record created in the section [above][record object]:
+```js
+const savePromise = skygear.publicDB.save(bob);
 ```
+`savePromise` is an [A+ Promise Object][] that either returns the record that was successfully saved or the error when the save operation failed. You can find out more about errors in the [Error Handling][] guide.
```

Comment:
- should be "`save` returns an.."?
- needs to mention the `save` function for `privateDB`
- the sample code needs to show the simplest form of error handling for easy understanding. It doesn't matter if we have it in every piece of codes